.github/workflows/pr-comment-build-dispatcher.yaml

```diff
@@ -49,19 +49,25 @@ jobs:
 
           $res
 
+      - name: Post a link to the workflow run to the PR
+        uses: mshick/add-pr-comment@v2
+        with:
+          issue: ${{ github.event.issue.pull_request.number }}
+          message: "Build action triggered: [${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})"
+
   update_sdk:
     needs: check_maintainer_membership
-    if: ( always() && needs.check_maintainer_membership.result == 'success'
-                   && contains(github.event.comment.body, '/update-sdk') )
+    if: ( needs.check_maintainer_membership.result == 'success'
+           && contains(github.event.comment.body, '/update-sdk') )
     name: "Build an updated SDK container"
     # SDK build needs access to bincache ssh secret
     secrets: inherit
     uses: ./.github/workflows/update-sdk.yaml
 
   build_image:
     needs: [ check_maintainer_membership, update_sdk ]
-    if: ( always() && needs.check_maintainer_membership.result == 'success'
-                   && (    contains(github.event.comment.body, '/build-image') || needs.update_sdk.result == 'success' ) )
+    if: ( needs.check_maintainer_membership.result == 'success'
```

Comment:
Apparently doing something like:

```yaml
if: (always() && !canceled()) && other_conditions
```

takes care of the "uncancel-able" job issue we saw when using ```always()```.

Reply:
OOOF 🙃
Let's give it a try without `always()` and see if it works. I'll keep the above in mind in case it doesn't.